docs/types.rst

```diff
@@ -426,8 +426,25 @@ function type should not return anything, the whole ``returns (<return types>)``
 part has to be omitted.
 
 By default, function types are internal, so the ``internal`` keyword can be
-omitted. In contrast, contract functions themselves are public by default,
-only when used as the name of a type, the default is internal.
+omitted. Note that this only applies to function types. Visibility has
+to be specified explicitly for functions defined in contracts, they
+do not have a default.
+
+A function type ``A`` is implicitly convertible to a function type ``B`` if and only if
+their parameter types are identical, their return types are identical,
+their internal/external property is identical and the state mutability of ``A``
+is not more restrictive than the state mutability of ``B``. In particular:
+``pure`` functions can be converted to ``view`` and ``non-payable`` functions and
```

Comment:
I had to read this three times to correctly understand the implicit parenthesis in the sentence. We should use full stops instead:
```
In particular: ``pure`` functions can be converted to ``view`` and ``non-payable`` functions.
``view`` functions can be converted to ``non-payable`` functions.
``payable`` functions can be converted to ``non-payable`` functions.
```

Apart from that I'm fine with it now.

Reply:
I will even make a list out of it!

Reply:
Also removed the sentence about what is least restrictive, it is just confusing.